File: site-keeper-data-access/src/main/java/com/igt/sitekeeper/requestmodel/ProjectRequestModel.java
package com.igt.sitekeeper.requestmodel;

import com.igt.sitekeeper.model.Feature;
import com.igt.sitekeeper.model.ProjectStatus;
import com.igt.sitekeeper.model.User;
import com.igt.sitekeeper.model.relationmodels.DisciplinesOnProjects;
import com.igt.sitekeeper.model.relationmodels.FeaturesOnProjects;
import lombok.Data;

import java.util.List;

@Data
public class ProjectRequestModel {
    private String startDate;
    private String endDate;
    private String name;
    private Long totalHours;
    private User projectManager;
    private String costingId;
    private List<Long> similarProjects;
    private List<FeaturesOnProjects> features;
    private List<DisciplinesOnProjects> disciplines;
    private ProjectStatus status;
}

File: site-keeper-frontend/src/models/Project/ProjectRequestModel.ts
import AlternateFeatureName from "../AlternateFeatureName"
import ProjectDiscipline from "../Disciplines/DisciplineRequestModel"
import UserModel from "../User/UserModel"
import FeatureModel from "../Feature/FeatureModel"
import ProjectStatus from "./ProjectStatus"

class ProjectRequestModel {
    startDate: string
    endDate: string
    name: string
    status: ProjectStatus | null
    totalHours: number
    costingId: string
    projectManager: UserModel | null
    similarProjects: number[]
    features: AlternateFeatureName[]
    disciplines: ProjectDiscipline[]

    constructor(startDate: string, endDate: string, name: string,
        status: ProjectStatus | null, totalHours: number, costingId: string, projectManager: UserModel | null, similarProjects: number[],
        features: AlternateFeatureName[], disciplines: ProjectDiscipline[]){
            this.startDate = startDate
            this.endDate = endDate
            this.name = name
            this.status = status
            this.totalHours = totalHours
            this.projectManager = projectManager
            this.similarProjects = similarProjects
            this.features = features
            this.disciplines = disciplines
            this.costingId = costingId
        }
}

export default ProjectRequestModel